profiling-ffi/src/profiles.rs

```diff
@@ -691,13 +694,22 @@ pub unsafe extern "C" fn ddog_prof_Profile_serialize(
             ))
         }
     };
+    let old_profile = match profile.reset(start_time.map(SystemTime::from)) {
```

Comment:
Probably should add something in the `ddog_prof_Profile_serialize` docs saying that the profile will be reset.

Reply:
https://github.com/DataDog/libdatadog/pull/242/files#diff-e1c96667d7e8245b5d270aa32f45b199929b795e3e26e7100a98d9e8125dbdc6R661